### DISCUSSION TITLE
pyright does not support python 2.7. how do I enable it?

### DISCUSSION THREAD

**Discussion by yurikoif** [microsoft/pyright#4528] [Q&A] [answered]
`pyright --pythonversion 2.7` prints `'2.7' is not a supported Python version; specify 3.3, 3.4, etc.`.
I have seen from previous posts that pyright actually works with python 2.7. could it be I have any problem with my environment?
`pyright --version` prints `pyright 1.1.281`.

**Comment by erictraut** [comment, marked as answer]
Pyright does not support Python 2.7, and there are no plans to support it.

**Reply by yurikoif** [reply]
thanks! but just curious why do i see posts like this https://github.com/microsoft/pyright/issues/829 suggesting 2.7 had been around?

**Reply by erictraut** [reply]
Pyright has never supported analysis of Python 2.x code. It was not designed or implemented with that in mind.
